ingest/src/main/java/com/microsoft/azure/kusto/ingest/resources/RankedStorageAccountSet.java

```diff
@@ -0,0 +1,60 @@
+package com.microsoft.azure.kusto.ingest.resources;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.microsoft.azure.kusto.ingest.utils.StreamUtils.roundRobinNestedList;
+
+public class RankedStorageAccountSet {
+    private final Map<String, RankedStorageAccount> accounts;
+
+    public RankedStorageAccountSet() {
+        this.accounts = new HashMap<>();
+    }
+
+    public void addResultToAccount(String accountName, boolean success) {
+        RankedStorageAccount account = accounts.get(accountName);
+        if (account != null) {
+            account.addResult(success);
+        }
+    }
+
+    public void addAccount(String accountName) {
+        if (!accounts.containsKey(accountName)) {
+            accounts.put(accountName, new RankedStorageAccount(accountName));
+        }
+    }
+
+    public void addAccount(RankedStorageAccount account) {
+        if (!accounts.containsKey(account.getAccountName())) {
+            accounts.put(account.getAccountName(), account);
+        }
+    }
+
+    @Nullable
+    public RankedStorageAccount getAccount(String accountName) {
+        return accounts.get(accountName);
+    }
+
+    public <T> List<T> getShuffledResources(Map<String, List<T>> resources) {
+        List<RankedStorageAccount> accounts = getShuffledAccounts();
+
+        // Get valid resources for each account
+        List<List<T>> validResources = accounts.stream().map(x -> resources.get(x.getAccountName())).filter(r -> r != null && !r.isEmpty())
```

Comment:
I have no idea what this works.
Also, this seems like a logic which ought to sit in ResourceManager

Reply:
There is a comment that explains it. 
I don't know how to make it simpler -> it loops over the shuffled account, for each one it looks up a resource in the resources map, and returns the valid ones.

Also, in c# the logic lives there.
You didn't want much logic in ResourceManager anyway.

Plus, this is just the most dead simple logic (just pick the valid ones).
When we will have a more complicated logic (taking into account the rank), it will be needed to be done here anyway.